exposee-ui/src/components/Registration/Registration..jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useContext } from 'react';
+//import { Link, useNavigate } from 'react-router-dom';
+import './Registration.css'
+import { UserContext } from '../../UserContext.js';
+
+const Registration = () => {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  const { updateUser } = useContext(UserContext);
+  //const navigate = useNavigate();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    try {
+      // Make the signup API request
+      const response = await fetch(`http://localhost:5173/users`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ username, email, password }),
+        credentials: 'include'
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        const loggedInUser = data.user;
+
+        console.log('Signup successful');
+
+        // Reset form fields
+        setUsername('');
+        setEmail('');
+        setPassword('');
+
+        // Update the user context
+        updateUser(loggedInUser);
+
+        // Navigate to the home page after successful login
+        navigate('/');
+      } else {
+        // Handle signup failure case
+        alert('Signup failed');
+      }
+    } catch (error) {
+      // Handle any network or API request errors
+      alert('Signup failed: ' + error);
+    }
+  };
+
+  return (
+    <div className="signup-form-container">
```

Comment:
I think this entire file is untested, it's called ..jsx
you should put it in the next PR